redshift/resource_redshift_schema.go

```diff
@@ -174,11 +618,122 @@ func resourceRedshiftSchemaCreate(db *DBConnection, d *schema.ResourceData) erro
+			query = fmt.Sprintf("%s PORT %d", query, port.(int))
+		}
+		iamRoleArnsRaw := d.Get(fmt.Sprintf("%s.%s", hiveMetastoreAttr, "iam_role_arns")).([]interface{})
+		iamRoleArns := []string{}
+		for _, arn := range iamRoleArnsRaw {
+			iamRoleArns = append(iamRoleArns, arn.(string))
+		}
+		query = fmt.Sprintf("%s IAM_ROLE '%s'", query, pqQuoteLiteral(strings.Join(iamRoleArns, ",")))
+	} else if _, isRdsPostgres := d.GetOk(rdsPostgresAttr); isRdsPostgres {
+		// rds postgres source
+		query = fmt.Sprintf("%s FROM POSTGRES DATABASE '%s'", query, pqQuoteLiteral(sourceDbName))
+		if sourceSchema, sourceSchemaIsSet := d.GetOk(fmt.Sprintf("%s.%s", rdsPostgresAttr, "schema")); sourceSchemaIsSet {
+			query = fmt.Sprintf("%s SCHEMA '%s'", query, pqQuoteLiteral(sourceSchema.(string)))
+		}
+		hostName := d.Get(fmt.Sprintf("%s.%s", rdsPostgresAttr, "hostname")).(string)
+		query = fmt.Sprintf("%s URI '%s'", query, pqQuoteLiteral(hostName))
+		if port, portIsSet := d.GetOk(fmt.Sprintf("%s.%s", rdsPostgresAttr, "port")); portIsSet {
+			query = fmt.Sprintf("%s PORT %d", query, port.(int))
+		}
+		iamRoleArnsRaw := d.Get(fmt.Sprintf("%s.%s", rdsPostgresAttr, "iam_role_arns")).([]interface{})
+		iamRoleArns := []string{}
+		for _, arn := range iamRoleArnsRaw {
+			iamRoleArns = append(iamRoleArns, arn.(string))
+		}
+		query = fmt.Sprintf("%s IAM_ROLE '%s'", query, pqQuoteLiteral(strings.Join(iamRoleArns, ",")))
+		secretArn := d.Get(fmt.Sprintf("%s.%s", rdsPostgresAttr, "secret_arn")).(string)
+		query = fmt.Sprintf("%s SECRET_ARN '%s'", query, pqQuoteLiteral(secretArn))
+	} else if _, isRdsMysql := d.GetOk(rdsMysqlAttr); isRdsMysql {
+		// rds mysql source
+		query = fmt.Sprintf("%s FROM MYSQL DATABASE '%s'", query, pqQuoteLiteral(sourceDbName))
+		hostName := d.Get(fmt.Sprintf("%s.%s", rdsMysqlAttr, "hostname")).(string)
+		query = fmt.Sprintf("%s URI '%s'", query, pqQuoteLiteral(hostName))
+		if port, portIsSet := d.GetOk(fmt.Sprintf("%s.%s", rdsMysqlAttr, "port")); portIsSet {
+			query = fmt.Sprintf("%s PORT %d", query, port.(int))
+		}
+		iamRoleArnsRaw := d.Get(fmt.Sprintf("%s.%s", rdsMysqlAttr, "iam_role_arns")).([]interface{})
+		iamRoleArns := []string{}
+		for _, arn := range iamRoleArnsRaw {
+			iamRoleArns = append(iamRoleArns, arn.(string))
+		}
+		query = fmt.Sprintf("%s IAM_ROLE '%s'", query, pqQuoteLiteral(strings.Join(iamRoleArns, ",")))
+		secretArn := d.Get(fmt.Sprintf("%s.%s", rdsMysqlAttr, "secret_arn")).(string)
+		query = fmt.Sprintf("%s SECRET_ARN '%s'", query, pqQuoteLiteral(secretArn))
+	} else if _, isRedshift := d.GetOk(redshiftAttr); isRedshift {
+		// redshift source
+		query = fmt.Sprintf("%s FROM REDSHIFT DATABASE '%s'", query, pqQuoteLiteral(sourceDbName))
+		if sourceSchema, sourceSchemaIsSet := d.GetOk(fmt.Sprintf("%s.%s", redshiftAttr, "schema")); sourceSchemaIsSet {
+			query = fmt.Sprintf("%s SCHEMA '%s'", query, pqQuoteLiteral(sourceSchema.(string)))
+		}
```

Comment:
Could you move each `if` body into separate function? It would make the code much more readable.

Example of what I mean:
```go
var configQuery string
if _, isDataCatalog := d.GetOk(dataCatalogAttr); isDataCatalog {
    configQuery = getDataCatalogConfigQueryPart(...)
} else if _, isHiveMetastore := d.GetOk(hiveMetastoreAttr); isHiveMetastore {
    configQuery = getHiveMetastoreConfigQueryPart(...)
} else if .....
```
and then just concatenate:

```go
query = fmt.Sprintf("%s %s", query, configQuery)
```

Reply:
sure thing

Reply:
okay I've pushed up this change. Was a quick fix but took a while to get to it.